V02/V03/V04/V05

vladn255 opened the pull request:
Для шагов заказа с 1 по 4 (без экрана подтверждения) сделал вёрстку. Поэтому некоторые интерактивные элементы и кнопки пока не работают, если это допустимо, конечно. Для навигации по вкладкам, после нажатия на кнопку "Забронировать" на главном экране, можно нажимать кнопки навигации по шагам (те, которые потом надо будет ограничивать).

По ходу работы столкнулся с парой проблем, которые поставили меня в тупик:
1) На 3 шаге (вкладка "Дополнительно") для чекбоксов нужно, чтобы галочка выходила за границы квадрата, но как это сделать? Если расположить галку поверх чекбокса (через z-index), то она будет загораживать собой label и не давать с ним взаимодействовать, что очевидно плохо. А если вытащить label (строки 11-17 в файле location-form.jsx) на уровень выше (т.е., сделать дополнительную обёртку), то это будет смотреться несколько несемантичненько, на мой взгляд. В каком направлении тут стоит двигаться?
2) На 2 шаге (вкладка "Модель") логика работы overflow: auto от меня безнадежно ускользнула. По идее (и судя по найденной информации в интернете), я всё делаю как надо - задаю высоту элементу .gallery через свободное пространство гридов (model-form.scss, строка 11), задаю overflow-y: auto (gallery.scss, строка 7), что не работает - контейнер беззаботно тянется по высоте. Проворачиваю аналогичный трюк со стоящим выше .model-form (строка 2), и всё работает. Но только до тех пор, пока у родителя order-page__content стоит overflow: hidden. Как они-то связаны?)) Есть идеи, что я делаю не так, и в чём может быть причина такого хаотичного поведения overflow?)

vladn255 committed:
V-02. Grid-layout

- markup;
- grid-layout;
- basic stylization.

vladn255 committed:
V-02/V-03/V-04/V-05 - desktop version

- finished markup for steps 1-4 (not 4.2);
- styles for steps 1-4 (desktop version).

vladn255 committed:
V-02/V-03/V-04/V-05

- finished stylization;
- basic navigation implemented

vladn255 committed:
V-02/V-03/V-04/V-05 - height fix

fix height adaptability issue

AlexsandrLitkevich commented on src/components/app/app.jsx:
```diff
@@ -1,11 +1,28 @@
         <div className="app">
-            <MainNav />
-            <Main />
+            <BrowserRouter>
```
BrowserRouter лучше использовать в файле index.js и оборачивать App

AlexsandrLitkevich commented on src/components/order-nav/order-nav.jsx:
```diff
@@ -0,0 +1,42 @@
+    )
+}
+
+const mapDispatchToProps = (dispatch) => ({
```
Почему используешь connect  а не хуки?

AlexsandrLitkevich commented on src/index.js:
```diff
@@ -1,16 +1,20 @@
-
-    <App />
-
+    <Provider store={store}>
```
вот тут и brawserrouter

AlexsandrLitkevich commented on src/sass/blocks/app.scss:
```diff
@@ -2,6 +2,7 @@
   min-height: $min-height-default;
+  height: 100vh;
   width: 100vw;
   max-width: $desktop-width;
```
У тебя приложение по ширине не во весь экран из за max-width и не понятно ты сначала ширину задаешь 100wh что говорит о ширине на весь экран и потом ограничиваешь max-width

AlexsandrLitkevich commented on src/store/action.js:
```diff
@@ -0,0 +1,15 @@
+const ActionType = {
```
тут зачем ты пишешь экшен а потом экшен крейтер  ????Напиши просто export const changeTab = (tab) =>{
type: ActionType.CHANGE_TAB,
        payload: tab
}

и сделай отдельный файл constants и туда записывай export const CHANGE_TAB = "CHANGE_TAB"

vladn255 committed:
post-review fixes

- overflow issue fix at model tab;
- checkbox icon overlapping fix;
- restructuring redux actions;
